lib/new_relic/agent/instrumentation/redis.rb

```diff
@@ -33,6 +34,10 @@
     NewRelic::Agent.logger.info('Installing Redis Instrumentation')
     if NewRelic::Agent::Instrumentation::Redis::Constants::HAS_REDIS_CLIENT
       RedisClient.register(NewRelic::Agent::Instrumentation::RedisClient::Middleware)
+
+      if defined?(Redis::Cluster::Client)
+        return RedisClient.register(NewRelic::Agent::Instrumentation::RedisClient::ClusterMiddleware)
```

Comment:
redis-clustering doesn't need `prepend` nor `chain` instrumentation, those methods are instrumented in the `ClusterMiddleware`.